### PR TITLE
Fuzzy search on bank name

### LINKED ISSUE #6: ACHDirectory Fuzzy search on Bank Name
If a 1-2-1 match is not found in the Name search and optional fuzzy search could be used to handle misspelling or other issues if the name was not found. 

A package such as [Fuzzy Search](https://github.com/lithammer/fuzzysearch) could be added to the Bank Name field for better results. 

Additional packages to be tested [closesmatch](https://github.com/schollz/closestmatch)

### PR DESCRIPTION
**Pull request by bkmoovio**
Closes #6

### CODE REVIEW

**Comment by bkmoovio** [comment]
Take a look let me know what can be adjusted.

**Comment by bkmoovio** [comment]
@wadearnold  @adamdecaf  There is a checksum mismatch error when downloading github.com/docker/docker v1.13.1.  I'm not sure why.

2.40s$ go test ./... -race -coverprofile=coverage.txt -covermode=atomic
go: downloading github.com/docker/docker v1.13.1
go: verifying github.com/docker/docker@v1.13.1: checksum mismatch
	downloaded: h1:IkZjBSIc8hBjLpqeAbeE5mca5mNgeatLHBy3GO78BWo=
	go.sum:     h1:5VBhsO6ckUxB0A8CE5LlUJdXzik9cbEbBTQ/ggeml7M=
The command "go test ./... -race -coverprofile=coverage.txt -covermode=atomic" exited with 1.

2.43s$ staticcheck ./...
go [list -e -json -compiled -test=true -export=false -deps=true -find=false -tags= -- ./...]: exit status 1: go: downloading github.com/docker/docker v1.13.1
go: verifying github.com/docker/docker@v1.13.1: checksum mismatch
	downloaded: h1:IkZjBSIc8hBjLpqeAbeE5mca5mNgeatLHBy3GO78BWo=
	go.sum:     h1:5VBhsO6ckUxB0A8CE5LlUJdXzik9cbEbBTQ/ggeml7M=
The command "staticcheck ./..." exited with 1

File: ACHDictionary.go [hunk 1]
```diff
@@ -7,16 +7,28 @@ package fed
...
+var (
+	// ACHJaroWinklerSimilarity is the search similarity percentage for strcmp.JaroWinkler for CustomerName
+	// (Financial Institution Name)
+	ACHJaroWinklerSimilarity = 0.85
+	// ACHLevenshteinSimilarity is the search similarity percentage for strcmp.Levenshtein for CustomerName
+	// (Financial Institution Name)
+	ACHLevenshteinSimilarity = 0.85
+	// ACHMaximumRecordsReturned is the maximum number of records to be returned - Currently this matches the FED Site
+	// maximum
+	ACHMaximumRecordsReturned = 499
```
Comment (adamdecaf) [inline ACHDictionary.go]: Were you thinking of supporting a `?limit=N` parameter on the HTTP api?

File: pkg/strcmp/strcmp.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+func JaroWinkler(a, b string) float64 {
+	if a == "" || b == "" {
+		return 0.00
+	}
+	return smetrics.JaroWinkler(a, b, 0.7, 4)
+}
+
+// Levenshtein is the "edit distance" between two strings. This is the count of operations
+// (insert, delete, replace) needed for two strings to be equal.
+func Levenshtein(a, b string) float64 {
```
Comment (adamdecaf) [inline pkg/strcmp/strcmp.go]: FYI, I changed this implementation a bit in OFAC.

https://github.com/moov-io/ofac/blob/v0.2.1/pkg/strcmp/strcmp.go#L24

**Comment by adamdecaf** [comment]
I've had that once before too. When it happened I just removed `go.sum` and had it regenerated.

File: pkg/strcmp/strcmp.go (hunk 2 above)
Comment (bkmoovio) [reply pkg/strcmp/strcmp.go]: I’ll take a look at the differences.

File: ACHDictionary.go (hunk 1 above)
Comment (bkmoovio) [reply ACHDictionary.go]: So let me know.  If there is a use case to return more, lets go though it.

**Comment by bkmoovio** [comment]
Seems to be a macOS issue:  The build passes for Linux

Adam logged an issue:

https://travis-ci.community/t/macos-go-tools-stopped-installing-binaries/2136

Uninstalled oclint to prevent interference with other packages.
If you need oclint, you must explicitly install it.
Services are not supported on osx
Updating gimme
git.checkout
$ git clone --depth=50 --branch=Fuzzy-search-on-Bank-Name https://github.com/moov-io/fed.git moov-io/fed
Cloning into 'moov-io/fed'...
remote: Enumerating objects: 65, done.
remote: Counting objects: 100% (65/65), done.
remote: Compressing objects: 100% (53/53), done.
remote: Total 328 (delta 15), reused 52 (delta 6), pack-reused 263
Receiving objects: 100% (328/328), 2.54 MiB | 14.19 MiB/s, done.
Resolving deltas: 100% (142/142), done.
$ cd moov-io/fed
$ git checkout -qf 3c9d38f399ca0440a470dba41c207e45641d6f6e
Setting environment variables from .travis.yml
$ export GO111MODULE=auto
0.26s$ travis_setup_go 1.11.x github.com/moov-io/fed
go version go1.11.5 darwin/amd64
$ export GO111MODULE=on
cache.1
Setting up build cache
$ export CASHER_DIR=${TRAVIS_HOME}/.casher
0.19s$ Installing caching utilities
0.01s3.54sattempting to download cache archive
fetching Fuzzy-search-on-Bank-Name/cache-osx-d5a04767d9fa6565dc7b2b45ae8b16595c2b991558053d4bf5b188d5c05f670a--go-1.11.x.tgz
found cache
0.01s4.00sadding /Users/travis/Library/Caches/go-build to cache
creating directory /Users/travis/Library/Caches/go-build
$ gimme version
v1.5.3
$ go version
go version go1.11.5 darwin/amd64
go.env
$ go env
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/travis/Library/Caches/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/travis/go"
GOPROXY=""
GORACE=""
GOROOT="/Users/travis/.gimme/versions/go1.11.5.darwin.amd64"
GOTMPDIR=""
GOTOOLDIR="/Users/travis/.gimme/versions/go1.11.5.darwin.amd64/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/travis/build/moov-io/fed/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/go-build172820637=/tmp/go-build -gno-record-gcc-switches -fno-common"
before_install.1
6.35s$ go get -u github.com/client9/misspell/cmd/misspell
go: finding github.com/schollz/closestmatch v2.1.0+incompatible
go: finding github.com/rickar/cal v1.0.0
go: finding github.com/xrash/smetrics v0.0.0-20170218160415-a3153f7040e9
go: finding github.com/moov-io/base v0.8.0
go: finding github.com/docker/docker v1.13.1
go: finding github.com/client9/misspell/cmd/misspell latest
go: finding github.com/client9/misspell/cmd latest
go: finding github.com/client9/misspell v0.3.4
go: downloading github.com/client9/misspell v0.3.4
before_install.2
6.73s$ go get -u golang.org/x/lint/golint
go: finding golang.org/x/lint/golint latest
go: finding golang.org/x/lint latest
go: downloading golang.org/x/lint v0.0.0-20181217174547-8f45f776aaf1
go: finding golang.org/x/tools/go/ast/astutil latest
go: finding golang.org/x/tools/go/gcexportdata latest
go: finding golang.org/x/tools/go/ast latest
go: finding golang.org/x/tools/go latest
go: finding golang.org/x/tools latest
go: downloading golang.org/x/tools v0.0.0-20190205181801-90c8b4f75bb8
before_install.3
7.72s$ go get -u honnef.co/go/tools/cmd/staticcheck
go: finding honnef.co/go/tools/cmd/staticcheck latest
go: finding honnef.co/go/tools/cmd latest
go: finding honnef.co/go/tools latest
go: downloading honnef.co/go/tools v0.0.0-20190128043916-71123fcbb8fe
go: finding github.com/BurntSushi/toml v0.3.1
go: downloading github.com/BurntSushi/toml v0.3.1
before_install.4
1.83s$ go get github.com/fzipp/gocyclo
go: finding github.com/fzipp/gocyclo latest
go: downloading github.com/fzipp/gocyclo v0.0.0-20150627053110-6acd4345c835
before_install.5
1.82s$ go get golang.org/x/tools/cmd/cover
go: finding golang.org/x/tools/cmd/cover latest
go: finding golang.org/x/tools/cmd latest
go: finding golang.org/x/tools latest
before_install.6
0.01s$ if [[ "$TRAVIS_OS_NAME" == "windows" ]]; then choco install -y mingw; export PATH=/c/tools/mingw64/bin:"$PATH"; fi
install
0.70s$ travis_install_go_dependencies 1.11.x -v
Using Go 1.11+ Modules
go: downloading github.com/xrash/smetrics v0.0.0-20170218160415-a3153f7040e9
go: downloading github.com/moov-io/base v0.8.0
go: downloading github.com/rickar/cal v1.0.0
before_script
0.02s$ GOFILES=$(find . -type f -name '*.go' | grep -v vendor)
0.01s$ if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then test -z $(gofmt -s -l $GOFILES); fi
The command "if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then test -z $(gofmt -s -l $GOFILES); fi" exited with 0.
71.32s$ go test ./... -race -coverprofile=coverage.txt -covermode=atomic
go: downloading github.com/docker/docker v1.13.1
ok  	github.com/moov-io/fed	4.866s	coverage: 92.8% of statements
ok  	github.com/moov-io/fed/pkg/strcmp	66.303s	coverage: 91.3% of statements
The command "go test ./... -race -coverprofile=coverage.txt -covermode=atomic" exited with 0.
0.01s$ misspell -error -locale US $GOFILES
/Users/travis/.travis/functions: line 104: misspell: command not found
The command "misspell -error -locale US $GOFILES" exited with 127.
0.02s$ gocyclo -over 25 $GOFILES
/Users/travis/.travis/functions: line 104: gocyclo: command not found
The command "gocyclo -over 25 $GOFILES" exited with 127.
0.01s$ golint -set_exit_status $GOFILES
/Users/travis/.travis/functions: line 104: golint: command not found
The command "golint -set_exit_status $GOFILES" exited with 127.
0.01s$ staticcheck ./...
/Users/travis/.travis/functions: line 104: staticcheck: command not found
The command "staticcheck ./..." exited with 127.
cache.2
store build cache
0.01s6.68schange detected (content changed, file is created, or file is deleted):
/Users/travis/Library/Caches/go-build/00/006baa34a97d5df34bb98e347567103b665adb9b8579e4db30a2d3c2add7a8d9-d
/Users/travis/Library/Caches/go-build/03/035fd280de12bdbc210601fb83418aa88bb4dfab92d1328da9ac5bce400be0f2-a
/Users/travis/Library/Caches/go-build/03/036f70bbcf8395a0a3a4dca12836f30a9fa73e44ac6c79454dc4e826cf025c8a-d
/Users/travis/Library/Caches/go-build/05/0516a7393cfa8dd2e827aa020397ba52ae3503440ae764f920fd9283cb570240-a
/Users/travis/Library/Caches/go-build/06/06847999b6aaebde1ad329c2fc899241f6aae4da45a40032f2273d768d4085d4-a
/Users/travis/Library/Caches/go-build/07/07dd008beb3ea24b05ae2fdf6ae90461828b4f961cbe50dcd52e1b1e3b4b4013-d
/Users/travis/Library/Caches/go-build/09/095ffcdb6515fd96f2624c26ad7d83576930b1234da9f50a92aa0307cbff6d5c-d
/Users/travis/Library/Caches/go-build/0b/0b80855455d56fe6552e4859f0c31555904a839e7ae777a934812828987f6e81-a
/Users/travis/Library/Caches/go-build/0b/0be9b6d37bf7999ef220177a748ca1f6c697cc4116a0c061c697c75fad8862fb-a
/Users/travis/Library/Caches/
...
changes detected, packing new archive
uploading Fuzzy-search-on-Bank-Name/cache-osx-d5a04767d9fa6565dc7b2b45ae8b16595c2b991558053d4bf5b188d5c05f670a--go-1.11.x.tgz
cache uploaded
Done. Your build exited with 1.

**Comment by bkmoovio** [comment]
adam [3:27 PM]
They enabled modules today and are looking at why macOS in specific broke


Brooke Kline [3:28 PM]
thanks Adam!

**Pull request merged**